### PR TITLE
Function-level optimization attributes.

### PR DESCRIPTION
**Pull request by eeckstein**
For now these are underscored attributes, i.e. compiler internal attributes:
    @_optimize(speed)
    @_optimize(size)
    @_optimize(none)
    
Those attributes override the command-line specified optimization mode for a specific function.
The @_optimize(none) attribute is equivalent to the already existing @_semantics("optimize.sil.never") attribute.

rdar://problem/35243425

### CODE REVIEW

**Comment by eeckstein** [comment]
@swift-ci test

**Comment by eeckstein** [comment]
@swift-ci smoke test

**Review by aschwaighofer — APPROVED** [review]
Looks good. Thank you.

**Comment by shahmishal** [comment]
ci.swift.org is down for update, I will re-trigger this PR once update has been completed.

**Comment by shahmishal** [comment]
@eeckstein  Due to merge conflict, I am not re-triggering the build.

File: include/swift/AST/Attr.def [hunk 1]
```diff
@@ -299,6 +299,10 @@ SIMPLE_DECL_ATTR(_implicitly_unwrapped_optional,
                  ImplicitlyUnwrappedOptional,
                  OnFunc | OnVar | OnSubscript | OnConstructor | RejectByParser, 72)
 
+DECL_ATTR(_optimize, Optimize,
+          OnFunc | OnConstructor | OnDestructor | OnSubscript |
```
Comment (slavapestov) [inline include/swift/AST/Attr.def]: Theoretically since you can apply this to subscripts, why not computed properties too?

**Comment by slavapestov** [comment]
@eeckstein This is a nice cleanup of the various flags, I like it. Can you replace existing usages of @_semantics("optimize.sil.never") in the test suite with @_optimize(none)? And why don't we get rid of the @_semantics variant altogether? It was never officially supported and from now on it's just unnecessary code to maintan.

**Comment by aschwaighofer** [comment]
@slavapestov We use @_semantics for "array semantics" we cannot get rid of it until we have an alternative.

**Comment by eeckstein** [comment]
@slavapestov Yes, let me do that in a follow-up commit

File: include/swift/AST/Attr.def (hunk 1 above)
Comment (eeckstein) [reply include/swift/AST/Attr.def]: I'll add it. This is OnVar, right?

**Comment by eeckstein** [comment]
Please test with following PR:
https://github.com/apple/swift-lldb/pull/256

@swift-ci smoke test

**Comment by eeckstein** [comment]
rdar://problem/35243425

**Pull request merged**